src/AppBundle/EventListener/I18nSecurityResponseListener.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AppBundle\EventListener;
+
+use Ma27\ApiKeyAuthenticationBundle\Event\AssembleResponseEvent;
+use Ma27\ApiKeyAuthenticationBundle\Ma27ApiKeyAuthenticationEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Listener for the response lifecycle of the apikey bundle.
+ * This listener transforms a deeper structure with translations.
+ *
+ * 'Message' -> ['de' => ['Meldung'], 'en' => ['Message']]
+ *
+ * This is useful for the i18n system of the frontend which needs translations of the security errors.
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+class I18nSecurityResponseListener implements EventSubscriberInterface
+{
+    /**
+     * @var mixed[]
+     */
+    private $locales = [];
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * Constructor.
+     *
+     * @param string[]            $locales
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(array $locales, TranslatorInterface $translator)
+    {
+        $this->locales    = array_keys($locales);
```

Comment:
I don't like this. If a service needs to work with the locales (e.g. creating a second parameter or using a ``keys`` expression for the parameter)